collections/users.go

```diff
@@ -31,21 +33,46 @@ func (u *Users) GetInstance() interface{} {
 }
 
 func (u *Users) GetIndexes() []*s.IndexConfig {
-	return []*s.IndexConfig{}
+	return []*s.IndexConfig{{
+		Path:   "DeviceID",
+		Unique: true,
+	}, {
+		Path: "ProjectID",
+	}}
 }
 
 func (u *Users) GetStoreID() *uuid.UUID {
 	return u.storeID
 }
 
-func (u *Users) Create(ctx context.Context, email string) (*User, error) {
+func (u *Users) GetOrCreate(ctx context.Context, projectID, deviceID string) (user *User, err error) {
 	ctx = AuthCtx(ctx, u.token)
-	user := &User{
-		Email:   email,
-		Teams:   []string{},
-		Created: time.Now().Unix(),
+	user, err = u.GetByDeviceID(ctx, deviceID)
+	if user != nil {
+		return
 	}
-	if err := u.threads.ModelCreate(ctx, u.storeID.String(), u.GetName(), user); err != nil {
+	if err != nil {
```

Comment:
Shouldn't the `err` be checked before `user`?

Reply:
Ah yeah, thanks! Previously, this was a get by ID, which errors if not found... but we didn't want to stop at that error. Fixed.